### PR TITLE
Add `VideoInterface` to conversion gallery

### PR DESCRIPTION
**Pull request by h-mayorquin**
As stated in the title. This has #181 as a base. Note that the MovieInterface was not using the context manager which we were using in all the other interfaces (`make_or_load_nwbfile`). This was added her e to keep consistent style access across the conversion gallery.

### CODE REVIEW

File: docs/conversion_examples_gallery/behavior/movie.rst [hunk 1]
```diff
@@ -0,0 +1,35 @@
+Video data conversion (multimedia formats)
+------------------------------------------
+
+Install NeuroConv with the additional dependencies necessary for reading multimedia data.
+
+.. code-block:: bash
+
+    pip install neuroconv[movie]
+
+This interface can handle conversions from avi, mov, mp4, wmv, flv and most FFmpeg supported formats to NWB using the
```
Comment (bendichter) [inline docs/conversion_examples_gallery/behavior/movie.rst:10]: link to ffmpeg

Comment (h-mayorquin) [reply docs/conversion_examples_gallery/behavior/movie.rst]: Done.

File: docs/conversion_examples_gallery/behavior/movie.rst [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+Install NeuroConv with the additional dependencies necessary for reading multimedia data.
+
+.. code-block:: bash
+
+    pip install neuroconv[movie]
+
+This interface can handle conversions from avi, mov, mp4, wmv, flv and most FFmpeg_ supported formats to NWB using the
+:py:class:`~neuroconv.datainterfaces.behavior.movie.moviedatainterface.MovieInterface` class.
+
+.. _FFmpeg: https://ffmpeg.org/
```
Comment (CodyCBakerPhD) [inline docs/conversion_examples_gallery/behavior/movie.rst:13]: Can you move this to the https://github.com/catalystneuro/neuroconv/blob/add_movie_interface_gallery/docs/conf.py#L90 area so all external links are in one place for easy searching/fixing? (tomorrow, I know you're done for today)

**Comment by CodyCBakerPhD** [comment]
LGTM once that external link is centralized

File: docs/conversion_examples_gallery/behavior/movie.rst (hunk 2 above)
Comment (h-mayorquin) [reply docs/conversion_examples_gallery/behavior/movie.rst:13]: I tried this for like 15 minutes and I could not figure out how to make it work. The documentation was not clear to me either:
https://www.sphinx-doc.org/en/master/usage/extensions/extlinks.html

I tried:
```

extlinks = {
    "nwbinspector": ("https://nwbinspector.readthedocs.io/en/dev/%s", ""),
    "FFmpeg": ("https://ffmpeg.org/%s", ""),
}

```
With the following signature:
```
:FFmpeg:`FFmpeg`
```

But this generates this:
image

File: docs/conversion_examples_gallery/behavior/movie.rst [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+.. code-block:: python
+
+    >>> from datetime import datetime
+    >>> from tabnanny import verbose
```
Comment (bendichter) [inline docs/conversion_examples_gallery/behavior/movie.rst:18]: what is the purpose of this import?

File: docs/conversion_examples_gallery/conversion_example_gallery.rst [hunk 4]
```diff
@@ -80,6 +80,7 @@ Behavior
 
     DeepLabCut <behavior/deeplabcut>
     SLEAP <behavior/sleap>
+    Multimedia Movies <behavior/movie>
```
Comment (bendichter) [inline docs/conversion_examples_gallery/conversion_example_gallery.rst:83]:
```suggestion
    Videos <behavior/movie>
```

**Comment by bendichter** [comment]
Can we change all of this from "movie" to "video"?

File: docs/conversion_examples_gallery/behavior/movie.rst (hunk 2 above)
Comment (CodyCBakerPhD) [reply docs/conversion_examples_gallery/behavior/movie.rst:13]: Just need the string to be :FFmpepg:`FFmpeg <>`.

Example on the Inspector: https://github.com/NeurodataWithoutBorders/nwbinspector/blame/dev/docs/best_practices/nwbfile_metadata.rst#L88 using https://github.com/NeurodataWithoutBorders/nwbinspector/blob/dev/docs/conf_extlinks.py#L26

**Comment by h-mayorquin** [comment]
> Can we change all of this from "movie" to "video"?

You mean all the mentions here in the conversion gallery or everywhere else?

File: docs/conversion_examples_gallery/behavior/movie.rst (hunk 3 above)
Comment (h-mayorquin) [reply docs/conversion_examples_gallery/behavior/movie.rst:18]: This is wrong. Thanks for spotting that.

File: docs/conversion_examples_gallery/behavior/movie.rst (hunk 2 above)
Comment (h-mayorquin) [reply docs/conversion_examples_gallery/behavior/movie.rst:13]: I tried that and it showed the `<>` on the document.

**Comment by bendichter** [comment]
> > Can we change all of this from "movie" to "video"?
> 
> You mean all the mentions here in the conversion gallery or everywhere else?

everywhere. I think "Video" is more accurate than "Movie"

File: docs/conversion_examples_gallery/behavior/movie.rst (hunk 3 above)
Comment (h-mayorquin) [reply docs/conversion_examples_gallery/behavior/movie.rst]: Removed.

**Comment by h-mayorquin** [comment]
> > > > > Can we change all of this from "movie" to "video"?
> > > > 
> > > > 
> > > > You mean all the mentions here in the conversion gallery or everywhere else?
> > > 
> > > 
> > > everywhere. I think "Video" is more accurate than "Movie"
> > 
> > 
> > I agree it is more accurate. The name of the interface as well?
> 
> yes. Can you smoothly deprecate that?

All right, on it.

**Comment by h-mayorquin** [comment]
The changes from Movie to Video are now in a separate PR to simplify diffs:
#190

**Comment by bendichter** [comment]
@CodyCBakerPhD the doctests aren't running here, but they should because we are adding a page to the gallery that needs to be tested.

**Comment by CodyCBakerPhD** [comment]
@bendichter Since this PR includes changes to `src`, the doctests are included in the main test battery rather than in isolation so we can include it on the coverage calculation.

image

The doctest only workflow only triggers for PRs where the _only_ files being modified are `docs`

**Comment by bendichter** [comment]
Thanks for the clarification!

**Comment by h-mayorquin** [comment]
I remember being confused by that (the doctest action) as well. Maybe we could change the name of the action to "doctest only"  or something better.

**Pull request merged**